### ISSUE TITLE
Should the current autocheck generator calls use 5?

### ISSUE THREAD

**Issue by sisuresh** [stellar/stellar-core#2680]
There are many calls in our testing code where we call `autocheck::generator` with the number 5. This is an investigative issue to determine if the current usage is good, or if it should be changed.

Example - https://github.com/stellar/stellar-core/blob/9500dc1d5e1182ca88694f1a226de7d8eec87941/src/ledger/test/LedgerTxnTests.cpp#L1227

**Comment by MonsieurNicolas** [comment]
Yeah I think that number controls the max recursion depth, so we may have to increase it with the work that was done recently, especially predicates.

**Comment by jonjove** [comment]
It also looks like it controls which branches of unions can get selected (at least under some circumstances).
